### PR TITLE
Updating with Angular 13 support

### PR DESCRIPTION
**Pull request by quinnjr**
- Updating dependencies to support Angular v13.
- Updating Jest.
- Updating Jest testing to work with Angular v13.
- Changing module output in root `tsconfig.json` to produce ES6 code.

NOTE:
I had to skip test 'should run inside Zone' at line 588 in
[packages/apollo-angular/tests/Apollo.spec.ts](packages/apollo-angular/tests/Apollo.spec.ts)
since `rxjs` no longer gives information on operator schedulers.

### Checklist:

- [ ] If this PR is a new feature, please reference a discussion where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests
- [ ] Try to include the Pull Request inside of CHANGELOG.md

### CODE REVIEW

**Comment by quinnjr** [comment]
See: https://github.com/kamilkisiela/apollo-angular/issues/1728

**Comment by quinnjr** [comment]
This pull request is still going to need a little work. I just looked at the built files and they aren't correctly identifying the exported modules.

In my opinion, these changes will also probably necessitate a major version bump and `ng-packagr` no longer exports UMD modules, only ESM modules.  I'm fairly certain Angular's `webpack` configuration supports ESM modules but I'm not entirely sure about backwards compatibility going back towards Angular v9.

File: .vscode/settings.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "**/.git": true,
     "**/.DS_Store": true,
     "**/node_modules/**": true,
-    "**/build/**": true,
+    "**/build/**": false,
```
Comment (kamilkisiela) [inline .vscode/settings.json:10]: why?

Comment (quinnjr) [reply .vscode/settings.json:10]: Not being able to see into the build directory caused me to miss all the changes that had happened with `ng-packagr` for their release for v13

**Comment by quinnjr** [comment]
This branch should be ready to go through the test suite now

**Comment by kondelik** [comment]
@gilgardosh Is there any (even rought) time estimate? (Like "next month" or "two weeks" or i dont know 😄 )

This is literally the last package preventing my project from updating to Angular 13

**Comment by quinnjr** [comment]
@kamilkisiela @gilgardosh Can we please get the workflows approved?

**Comment by gilgardosh** [comment]
It needs some fixes for tests, but rather than that looks like it should work just fine.
@quinnjr after you fix it, I guess merge will be very quick. this week even.

**Comment by quinnjr** [comment]
@gilgardosh Test should be running correctly now

Screenshot from 2021-12-01 07-53-18

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -74,15 +74,15 @@ jobs:
...
+        run: yarn test --ci
 
   angular:
     name: Test on Angular ${{ matrix.angular_version }}
     needs: build
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        angular_version: [9, 10, 11, 12]
+        angular_version: [11, 12, 13]
```
Comment (kamilkisiela) [inline .github/workflows/main.yml:85]:
```suggestion
        angular_version: [12, 13]
```

File: packages/apollo-angular/package.json [hunk 3]
```diff
@@ -28,16 +30,16 @@
...
     "build:schematics": "tsc -p schematics/tsconfig.json && node ./scripts/move-schematics.js",
-    "test": "jest --config jest.config.js",
+    "test": "NODE_OPTIONS=--experimental-modules jest --config jest.config.js",
     "prebuild": "rm -rf build/",
     "deploy": "yarn build && npm publish build",
     "deploy:next": "yarn build && npm publish build --tag next"
   },
   "peerDependencies": {
-    "@angular/core": "^9.0.0 || ^10.0.0 || ^11.0.0 || ^12.0.0",
+    "@angular/core": "^9.0.0 || ^10.0.0 || ^11.0.0 || ^12.0.0 || ^13.0.0",
```
Comment (kamilkisiela) [inline packages/apollo-angular/package.json:39]:
```suggestion
    "@angular/core": "^12.0.0 || ^13.0.0",
```

File: packages/apollo-angular/package.json [hunk 4]
```diff
@@ -28,16 +30,16 @@
...
     "deploy": "yarn build && npm publish build",
     "deploy:next": "yarn build && npm publish build --tag next"
   },
   "peerDependencies": {
-    "@angular/core": "^9.0.0 || ^10.0.0 || ^11.0.0 || ^12.0.0",
+    "@angular/core": "^12.0.0 || ^13.0.0",
     "@apollo/client": "^3.0.0",
     "graphql": "^14.0.0 || ^15.0.0",
-    "rxjs": "^6.0.0",
+    "rxjs": "^6.0.0 || ^7.0.0",
```
Comment (kamilkisiela) [inline packages/apollo-angular/package.json:42]: @gilgardosh

**Comment by anymos** [comment]
Hi there, any idea when this merge will be available on npm 'officially' ?

**Pull request merged**